### PR TITLE
Jetpack Free in most popular plans experiment

### PR DESCRIPTION
**Pull request by monsieur-z**
### Changes proposed in this Pull Request

This PR implements the new pricing page experiment. It moves Jetpack Free to the most popular section, and CRM Free to the bottom of the page.

Fixes 1196341175636977-as-1199527862646261 

### Testing instructions

- Download the PR and run Calypso and cloud concurrently
- For both platforms, check the following
- Select the control version (`withoutFree_control`) of the experiment (`jetpackPopularPlans`)
- Check that the plans/pricing page looks and behaves like production
- Select the test version (`withFree_test`) of the experiment
- Check the order of the most popular section (Free, Security Daily, Complete)
- Check the order of the more products section (Backup Daily, Backup Realtime, Security Realtime, etc.)
- Check the content of the Jetpack Free card against the mockups (see linked task)
- Check that the CRM Free card takes the full width of the cards container
- Check the content of the CRM Free card against the mockups

### Screenshots

_Jetpack Free is part of the most popular section_
Screen Shot 2021-03-03 at 3 33 43 PM

_Backpu Daily is moved to the next section_
Screen Shot 2021-03-03 at 3 33 50 PM

_CRM Free is moved to the bottom of the page_
Screen Shot 2021-03-03 at 3 34 01 PM

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/i7/jetpack-crm-card.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+import './style.scss';
+
+const JetpackCrmCard: FC = () => {
+	const translate = useTranslate();
+
+	return (
+		<div className="jetpack-crm-card jetpack-product-card" data-e2e-product-slug="free">
+			<header className="jetpack-crm-card__header">
+				<h3 className="jetpack-crm-card__title">{ translate( 'CRM Free' ) }</h3>
```
Comment (a8ci18n) [inline client/my-sites/plans/jetpack-plans/i7/jetpack-crm-card.tsx:19]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'Jetpack CRM Free' )` *ES Score: 7*

File: client/my-sites/plans/jetpack-plans/i7/jetpack-crm-card.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+			<ul className="jetpack-crm-card__features-list">
+				{ [
+					{
+						text: translate( 'Unlimited contacts' ),
+					},
+					{
+						text: translate( 'Manage billing and create invoices' ),
+					},
+					{
+						text: translate( 'Fully integrated with WordPress' ),
```
Comment (a8ci18n) [inline client/my-sites/plans/jetpack-plans/i7/jetpack-crm-card.tsx:41]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'CRM fully integrated with WordPress' )` *ES Score: 9*

File: client/components/jetpack/card/jetpack-free-card/button.tsx [hunk 3]
```diff
@@ -1,92 +1,40 @@
...
-const JetpackFreeCardButton: FC< JetpackFreeCardButtonProps > = ( {
+const JetpackFreeCardButton: React.FC< Props > = ( {
 	className,
 	label,
 	primary = false,
 	siteId,
 	urlQueryArgs,
 } ) => {
 	const translate = useTranslate();
-	const wpAdminUrl = useSelector( getJetpackWpAdminUrl );
```
Comment (monsieur-z) [inline client/components/jetpack/card/jetpack-free-card/button.tsx:42]: Moving this logic to a hook.

File: client/components/jetpack/card/jetpack-product-card/index.tsx [hunk 4]
```diff
@@ -104,7 +113,7 @@ const DisplayPrice = ( {
 		);
 	}
 
-	if ( isFree ) {
+	if ( isFree && ! isJetpackFree ) {
```
Comment (monsieur-z) [inline client/components/jetpack/card/jetpack-product-card/index.tsx:116]: Checking the product slug in this component is not scalable, but given this is an experiment it's the cleanest way of doing for now. We can revisit this depending on the experiment outcome.

File: client/lib/abtest/active-tests.js [hunk 5]
```diff
@@ -59,4 +59,13 @@ export default {
 		defaultVariation: 'control',
 		allowExistingUsers: false,
 	},
+	jetpackPopularPlans: {
+		datestamp: '20210304',
```
Comment (monsieur-z) [inline client/lib/abtest/active-tests.js]: We might need to update the date before launching.

File: client/lib/plans/plans-list.js [hunk 6]
```diff
@@ -932,52 +932,22 @@ export const PLANS_LIST = {
 	},
 
 	[ constants.PLAN_JETPACK_FREE ]: {
```
Comment (monsieur-z) [inline client/lib/plans/plans-list.js:944]: I checked that the copy below is not shown anywhere anymore.

File: client/lib/plans/plans-list.js [hunk 7]
```diff
@@ -932,52 +932,22 @@ export const PLANS_LIST = {
...
-		term: constants.TERM_ANNUALLY,
```
Comment (monsieur-z) [inline client/lib/plans/plans-list.js:945]: Removing the term prevents us from creating a new entry for the monthly term, which seemed confusing given the plan is free.

File: client/my-sites/plan-price/index.jsx [hunk 8]
```diff
@@ -33,13 +33,13 @@ export class PlanPrice extends Component {
...
 
-		if ( ! currencyCode || ! rawPrice ) {
+		if ( ! currencyCode || typeof rawPrice !== 'number' ) {
 			return null;
 		}
 
 		// "Normalize" the input price or price range.
 		const rawPriceRange = Array.isArray( rawPrice ) ? rawPrice.slice( 0, 2 ) : [ rawPrice ];
-		if ( rawPriceRange.includes( 0 ) ) {
+		if ( rawPrice > 0 && rawPriceRange.includes( 0 ) ) {
```
Comment (monsieur-z) [inline client/my-sites/plan-price/index.jsx:42]: Necessary to display 0.

File: client/my-sites/plans/jetpack-plans/utils.ts [hunk 9]
```diff
@@ -343,7 +348,7 @@ function slugIsJetpackProductSlug( slug: string ): slug is JetpackProductSlug {
 	return slug in JETPACK_PRODUCTS_LIST;
 }
 function slugIsJetpackPlanSlug( slug: string ): slug is JetpackPlanSlugs {
-	return [ ...JETPACK_LEGACY_PLANS, ...JETPACK_RESET_PLANS ].includes( slug );
+	return JETPACK_PLANS.includes( slug );
```
Comment (monsieur-z) [inline client/my-sites/plans/jetpack-plans/utils.ts:351]: Same array, plus Jetpack Free.

File: client/components/jetpack/card/jetpack-free-card/use-jetpack-free-button-props.ts [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+
+	// `siteId` is going to be null if the user is not logged in, so we need to check
+	// if there is a site in the URL also
+	const isSiteinContext = siteId || site;
+
+	// Jetpack Connect flow uses `url` instead of `site` as the query parameter for a site URL
+	const { site: url, ...restQueryArgs } = urlQueryArgs;
+
+	return isJetpackCloud() && ! isSiteinContext
+		? addQueryArgs( { url, ...restQueryArgs }, `https://wordpress.com${ JPC_PATH_BASE }` )
```
Comment (rcanepa) [inline client/components/jetpack/card/jetpack-free-card/use-jetpack-free-button-props.ts]: We can't forget to add here the `plan=jetpack_free` query parameter introduced in #50773.

File: client/my-sites/plans/jetpack-plans/product-grid/index.tsx [hunk 11]
```diff
@@ -42,6 +44,8 @@ import type { JetpackPlanSlugs } from 'calypso/lib/plans/types';
  */
 import './style.scss';
 
+const MAX_CARDS_PER_ROW = 3;
```
Comment (mattgawarecki) [inline client/my-sites/plans/jetpack-plans/product-grid/index.tsx:47]: This is a really small change, but it makes the wrapping logic so much easier to understand. Thank you!

File: client/my-sites/plans/jetpack-plans/product-grid/index.tsx [hunk 12]
```diff
@@ -206,11 +217,7 @@ const ProductGrid: React.FC< ProductsGridProps > = ( {
 						</li>
 					) ) }
 				</ul>
-				<div className="product-grid__free">
-					{ ( isInConnectFlow || ( isInJetpackCloud && ! isSiteInContext ) ) && (
-						<JetpackFreeCard siteId={ siteId } urlQueryArgs={ urlQueryArgs } />
-					) }
-				</div>
+				<div className="product-grid__free">{ bottomCard }</div>
```
Comment (mattgawarecki) [inline client/my-sites/plans/jetpack-plans/product-grid/index.tsx]: Can we conditionally render the whole `div` here? Ex:
```suggestion
				{ bottomCard && <div className="product-grid__free">{ bottomCard }</div> }
```

File: client/lib/abtest/active-tests.js (hunk 5 above)
Comment (mattgawarecki) [reply client/lib/abtest/active-tests.js]: Yeah, let's go ahead and change this to Monday:

```suggestion
		datestamp: '20210308',
```

File: client/components/jetpack/card/jetpack-free-card/use-jetpack-free-button-props.ts (hunk 10 above)
Comment (mattgawarecki) [reply client/components/jetpack/card/jetpack-free-card/use-jetpack-free-button-props.ts]: Good catch Renzo! +1 to this ☝️

File: client/components/jetpack/card/jetpack-free-card/use-jetpack-free-button-props.ts [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+		: wpAdminUrl || jetpackAdminUrlFromQuery || JPC_PATH_BASE;
+};
+
+export default function useJetpackFreeButtonProps(
```
Comment (mattgawarecki) [inline client/components/jetpack/card/jetpack-free-card/use-jetpack-free-button-props.ts:77]: I _really_ like the idea of using a hook here to cut down on noise in other parts of the code. Good thinking on this, Kevin 👍

**Review by mattgawarecki — APPROVED** [review]
Everything tested well for me in all experiences and screen sizes. Unit tests all pass, and code looks reasonable. I think it's ready for launch when the time is right! 👍

**Pull request merged**